forte/data/extractor/base_extractor.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from abc import ABC
+from typing import Dict, Any, Union, Type
+from ft.onto.base_ontology import Annotation
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.vocabulary import Vocabulary
+from forte.data.converter.feature import Feature
+
+
+class WrapperClass:
+    def __init__(self):
+        self.obj = None
+        self.func_name = None
+
+    def __call__(self, obj, func_name):
+        self.obj = obj
+        self.func_name = func_name
+        return self.wrapper
+
+    def wrapper(self, *args, **kwargs):
+        assert self.obj, """When vocab_mehtod is raw,
+        vocabulary is not built and operation on vocabulary should not
+        be called."""
+        return getattr(self.obj, self.func_name)(*args, **kwargs)
+
+
+class BaseExtractor(ABC):
+    '''This class is used to get Feature from Datapack and
+    add prediction back to Datapack.
+    '''
+    def __init__(self, config: Union[Dict, Config]):
+        '''Config: {"entry_type" : required, Type[Annotation],
```

Comment:
I guess the indentations need to be adjusted and the documentation explaining each configuration can be explained here?

Reply:
Can you show me one example? I will fix this.

Reply:
For code-like comments, useing code-block will be better

https://sublime-and-sphinx-guide.readthedocs.io/en/latest/code_blocks.html

Reply:
I think my previous comment is not correct. But anyway, I have found a reasonable way to write docstring. Please refer to my [PR](https://github.com/asyml/forte/pull/297/files#diff-bf1fae29dabdb8630c19ce4bd6033b7b65608fdf5e32b63d270f212519841b0fR34) for the style (for example, the docstring for `__init__` should be above `__init__` as this is the class docstring).